### PR TITLE
Adding support for withPidMode #332

### PR DESCRIPTION
**Pull request by alex-glv**

### CODE REVIEW

File: src/main/java/com/github/dockerjava/api/model/HostConfig.java [hunk 1]
```diff
@@ -75,13 +75,16 @@
     @JsonProperty("CpuShares")
     private int cpuShares = 0;
 
+    @JsonProperty("PidMode")
+    private String pidMode;
+
     public HostConfig() {
     }
 
     public HostConfig(Bind[] binds, Link[] links, LxcConf[] lxcConf, LogConfig logConfig, Ports portBindings,
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/api/model/HostConfig.java]: @marcuslinke do you need at all this huge constructor?

File: src/main/java/com/github/dockerjava/api/model/HostConfig.java [hunk 2]
```diff
@@ -75,13 +75,16 @@
...
             boolean publishAllPorts, boolean privileged, boolean readonlyRootfs, String[] dns, String[] dnsSearch,
             VolumesFrom[] volumesFrom, String containerIDFile, Capability[] capAdd, Capability[] capDrop,
-            RestartPolicy restartPolicy, String networkMode, Device[] devices, String[] extraHosts, Ulimit[] ulimits) {
+            RestartPolicy restartPolicy, String networkMode, Device[] devices, String[] extraHosts, Ulimit[] ulimits, String pidMode) {
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/api/model/HostConfig.java]: please put on new line as width become > 120

File: src/main/java/com/github/dockerjava/api/model/HostConfig.java [hunk 3]
```diff
@@ -194,6 +198,8 @@ public int getCpuShares() {
         return cpuShares;
     }
 
+    public String getPidMode() { return pidMode; }
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/api/model/HostConfig.java]: don't inline content

File: src/test/java/com/github/dockerjava/core/command/CreateContainerCmdImplTest.java [hunk 4]
```diff
@@ -1,5 +1,14 @@
 package com.github.dockerjava.core.command;
 
+import com.github.dockerjava.api.ConflictException;
+import com.github.dockerjava.api.DockerException;
+import com.github.dockerjava.api.command.CreateContainerResponse;
+import com.github.dockerjava.api.command.InspectContainerResponse;
+import com.github.dockerjava.api.model.*;
+import com.github.dockerjava.client.AbstractDockerClientTest;
+import org.testng.ITestResult;
+import org.testng.annotations.*;
```
Comment (KostyaSha) [inline src/test/java/com/github/dockerjava/core/command/CreateContainerCmdImplTest.java]: don't use `*`

File: src/test/java/com/github/dockerjava/core/command/CreateContainerCmdImplTest.java [hunk 5]
```diff
@@ -543,4 +546,6 @@ public void createContainerWithLogConfig() throws DockerException {
         // null becomes empty string
         assertEquals(inspectContainerResponse.getHostConfig().getLogConfig().type, logConfig.type);
     }
+
```
Comment (KostyaSha) [inline src/test/java/com/github/dockerjava/core/command/CreateContainerCmdImplTest.java]: remote unrelated change

File: src/main/java/com/github/dockerjava/api/command/CreateContainerCmd.java [hunk 6]
```diff
@@ -134,6 +134,8 @@
 
     public boolean isTty();
 
+    public String getPidMode();
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/api/command/CreateContainerCmd.java]: Annotate with `@CheckForNull`

File: src/main/java/com/github/dockerjava/api/model/HostConfig.java (hunk 3 above)
Comment (alex-glv) [inline src/main/java/com/github/dockerjava/api/model/HostConfig.java]: This and other mentioned issues are fixed in subsequent commits.

File: src/main/java/com/github/dockerjava/api/model/HostConfig.java [hunk 7]
```diff
@@ -194,6 +199,10 @@ public int getCpuShares() {
         return cpuShares;
     }
 
+    public String getPidMode() {
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/api/model/HostConfig.java]: `@CheckForNull` here also required as it doesn't inherited by any interfaces

File: src/main/java/com/github/dockerjava/api/model/HostConfig.java (hunk 1 above)
Comment (marcuslinke) [inline src/main/java/com/github/dockerjava/api/model/HostConfig.java]: Nope. Not needed. It's already removed in 'issue-246' branch. So no need to remove it here.

**Comment by alex-glv** [comment]
I think all issues but humongous constructor were addressed. Let me know if I missed anything.
Thanks.

**Comment by marcuslinke** [comment]
Thanks for your contribution! Hopefully I'm able to merge this evening.

File: src/main/java/com/github/dockerjava/core/command/CreateContainerCmdImpl.java [hunk 8]
```diff
@@ -733,4 +738,11 @@ public CreateContainerCmdImpl withWorkingDir(String workingDir) {
         return this;
     }
 
+    @Override
+    public CreateContainerCmd withPidMode(String pidMode) {
```
Comment (mwl) [inline src/main/java/com/github/dockerjava/core/command/CreateContainerCmdImpl.java]: If you want to reflect the CLI, it should me `withPid(String pidMode)`:

```
  --pid=                          PID namespace to use
```

Comment (alex-glv) [inline src/main/java/com/github/dockerjava/core/command/CreateContainerCmdImpl.java]: Good catch! Thanks.
